src/Game.ts

```diff
@@ -39,11 +38,24 @@ export default class Game {
         return this.prevQuestDescription;
     }
 
-    // returns all available resource conversions in the order in which they will be applied
-    getResourceConversions(): Conversion[] {
-        const allConversions: Conversion[] = Arrays.flatten(this.world.getTiles().map((tile: Tile) => tile.resourceConversions));
+    // Returns all available free and costly resource conversions in the order
+    // in which they will be applied, as two separate arrays
+    getResourceConversions(): { free: Conversion[]; costly: Conversion[]; } {
+        const allConversions: Conversion[] = Arrays.flatten(this.world.getTiles().map(tile => tile.resourceConversions));
```

Comment:
Its up to you but if it were me I would consider calling `this.getAllResourceConversions()` and then partitioning, to avoid duplicating the code that iterates over the world tiles. Since partition maintains the order, you then wouldn't have to sort afterwards because the two subsequences would already each be sorted. Sorting and then partitioning would be a bit slower than partitioning and then sorting but I don't think it should matter.

I trust your judgement on this one, though.

Reply:
That's just obviously less efficient in order to save one duplicated line of otherwise very clear code. I can extract that line to a private method if you prefer.

Reply:
You're right, the way you're doing it now is better.